Readme.md

```diff
@@ -1,30 +1,51 @@
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T260978)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
-
-<!-- default file list -->
-*Files to look at*:
-
-* [Default.aspx](./CS/Default.aspx) (VB: [Default.aspx](./VB/Default.aspx))
-* [Default.aspx.cs](./CS/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/Default.aspx.vb))
-<!-- default file list end -->
-# How to use ASPxRichEdit to edit RTF data in ASPxGridView's EditForm
+# Grid View for ASP.NET Web Forms - How to use the Rich Text Editor to edit RTF data in the Edit Form
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/t260978/)**
 <!-- run online end -->
 
+This example demonstrates how to add the [Rich Text Editor](https://docs.devexpress.com/AspNet/17721/components/rich-text-editor) to the [Grid View](https://docs.devexpress.com/AspNet/5823/components/grid-view) control's edit form. The Rich Text Editor displays RTF data from the data source bound to the Grid View and allows users to modify this data.
+
+> **Note**  
+> Data modification is not allowed in an online version of this example. To allow editing, download the example and comment out the [throw](./CS/Default.aspx.cs#L97) expression in the `RowUpdating` event handler.
+
+## Overview
+
+Follow the steps below to add the Rich Text Editor in the Grid View's edit form:
+
+1. Create the [Grid View](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) control, populate it with columns, and bind the control to a data source.
 
-This example is based on the <a href="https://www.devexpress.com/Support/Center/p/E4257">How to use ASPxHtmlEditor to edit RTF data</a> one. It illustrates how to accomplish the same task using a new ASPxRichEdit control instead of the ASPxHtmlEditor one
+2. Add the [Rich Text Editor](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit) control to the [EditItemTemplate](https://docs.devexpress.com/AspNet/DevExpress.Web.GridViewDataColumn.EditItemTemplate?p=netframework) of the column that contains RTF data.
+
+3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event. If the event occurs when a new row is being edited, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. If an existing row is being edited, get RTF data from the corresponding cell and pass the data and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method. 
```

Comment:
From this paragraph, I cannot understand whether this event occurs when a user activates an in-place cell editor/row editor or the event occurs when a row is being edited for some time. Right now, your wording suggests the latter. More clarity here is in order. In my opinion, this event occurs when a user activates an in-place editor, and the editor itself is about to become active. That is, the editing process is about to start at the moment of this event occurrence.

Reply:
When a user clicks the **New** or **Edit** button in the command column, the edit form appears. This form contains the Rich Text Editor, so after the edit form appears, the Rich Text Editor's `Init` event occurs. To check whether a user clicked the **New** or **Edit** button, the example uses the Grid View's IsNewRowEditing property, what's why I say "is being edited". However, the row is rather about to being edited.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) to determine whether the edit form after a user clicked the **New** or **Edit** button. If the user cliked **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get RTF data from the corresponding cell and pass the data and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method. 
```

Reply:
When a user clicks the **New** or **Edit** button in the command column, the edit form appears. This form contains the Rich Text Editor, so after the edit form appears, the Rich Text Editor's `Init` event occurs. To check whether a user clicked the **New** or **Edit** button, the example uses the Grid View's IsNewRowEditing property, what's why I say "is being edited". However, the row is rather about to being edited.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) to determine whether the edit form after a user clicked the **New** or **Edit** button. If the user cliked **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get RTF data from the corresponding cell and pass the data and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method. 
```

Reply:
Yes. Your explanations make sense indeed. Thank you. However, I suggest a number of adjustments for your latest version:

> cliked

A typo.

> determine

When you say "determine" -- it more says of modification -- you determine its state, appearance, that is, actively affect it. A synonym for "determine" is "specify", for example. However, in this case, you just need to know what it is. Therefore, you "identify", that is, learn something.

> Otherwise, get formatted text string from the corresponding cell

Since "string" is countable, you've missed an article.

```suggestion
3. Handle the Rich Text Editor control's [Init](https://learn.microsoft.com/en-us/dotnet/api/system.web.ui.control.init?view=netframework-4.8.1) event that occurs after the edit form appears. Use the Grid View's [IsNewRowEditing](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView.IsNewRowEditing?p=netframework) property to identify if the edit form appeared in response to a click on the **New** or **Edit** button. If a user clicked the **New** button, assign a new unique identifier to the Rich Text Editor's [DocumentId](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.DocumentId?p=netframework) property. Otherwise, get a formatted text string from the corresponding cell and pass the string and the row key to the control's [Open](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxRichEdit.ASPxRichEdit.Open(System.String-DevExpress.XtraRichEdit.DocumentFormat-System.Func-System.Byte---)) method.
```